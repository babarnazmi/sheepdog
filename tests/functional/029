#!/bin/bash

# Test sheep VDI redundancy level

. ./common

for i in `seq 0 22`; do
	_start_sheep $i;
done

_wait_for_sheep 23

_cluster_format -c 3

for i in `seq 2 4`; do
	$DOG vdi create test$i 20M -c $i -P
done

$DOG vdi create -c 4:2 test5 20M -P
$DOG vdi create -c 4:3 test6 20M -P
$DOG vdi create -c 4:4 fail 20M -P
$DOG vdi create -c 4:5 fail 20M -P
$DOG vdi create -c 16:7 test7 20M -P
$DOG vdi create -c 16:8 fail 20M -P
$DOG vdi snapshot -s tag test2
$DOG vdi clone -s 1 test2 clone -c 3

for i in `seq 2 3`; do
	_kill_sheep $i;
done

for i in `seq 23 24`; do
	_start_sheep $i;
done

_wait_for_sheep_recovery 0

$DOG vdi delete clone
$DOG vdi delete -s 1 test2

_vdi_list

for i in `seq 2 7`; do
	$DOG vdi object test$i -i 1;
done

