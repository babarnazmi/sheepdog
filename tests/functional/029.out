QA output created by 029
using backend plain store
Invalid parameter 4:4
To create erasure coded vdi, set -c x:y
  x(2,4,8,16)  - number of data strips
  y(1 to 15)   - number of parity strips
and meet the condition x > y
To create replicated vdi, set -c x
  x(0 < x <= 31) - number of replicated copies
Invalid parameter 4:5
To create erasure coded vdi, set -c x:y
  x(2,4,8,16)  - number of data strips
  y(1 to 15)   - number of parity strips
and meet the condition x > y
To create replicated vdi, set -c x
  x(0 < x <= 31) - number of replicated copies
There are not enough nodes(23) to hold the copies(24)
Usage: dog vdi create [-P] [-c copies] [-a address] [-p port] [-h] [-r] [-v] <vdiname> <size>
Options:
  -P, --prealloc          preallocate all the data objects
  -c, --copies            specify the data redundancy level
  -a, --address           specify the daemon address (default: localhost)
  -p, --port              specify the daemon port
  -h, --help              display this help and exit
  -r, --raw               raw output mode: omit headers, separate fields with
                          single spaces and print all sizes in decimal bytes
  -v, --verbose           print more information than default
  Name        Id    Size    Used  Shared    Creation time   VDI id  Copies  Tag
  test5        0   20 MB   20 MB  0.0 MB DATE   fd2c30     6              
  test4        0   20 MB   20 MB  0.0 MB DATE   fd2de3     4              
  test7        0   20 MB   20 MB  0.0 MB DATE   fd2f96    23              
  test6        0   20 MB   20 MB  0.0 MB DATE   fd3149     7              
  test3        0   20 MB   20 MB  0.0 MB DATE   fd3662     3              
  test2        0   20 MB  0.0 MB   20 MB DATE   fd3816     2              
Looking for the object 0xfd381500000001 (the inode vid 0xfd3816 idx 1) with 23 nodes

127.0.0.1:7000 doesn't have the object
127.0.0.1:7001 doesn't have the object
127.0.0.1:7004 doesn't have the object
127.0.0.1:7005 doesn't have the object
127.0.0.1:7006 doesn't have the object
127.0.0.1:7007 doesn't have the object
127.0.0.1:7008 doesn't have the object
127.0.0.1:7009 doesn't have the object
127.0.0.1:7010 doesn't have the object
127.0.0.1:7011 doesn't have the object
127.0.0.1:7012 doesn't have the object
127.0.0.1:7013 doesn't have the object
127.0.0.1:7014 doesn't have the object
127.0.0.1:7015 doesn't have the object
127.0.0.1:7016 has the object (should be 2 copies)
127.0.0.1:7017 doesn't have the object
127.0.0.1:7018 doesn't have the object
127.0.0.1:7019 doesn't have the object
127.0.0.1:7020 doesn't have the object
127.0.0.1:7021 doesn't have the object
127.0.0.1:7022 has the object (should be 2 copies)
127.0.0.1:7023 doesn't have the object
127.0.0.1:7024 doesn't have the object
Looking for the object 0xfd366200000001 (the inode vid 0xfd3662 idx 1) with 23 nodes

127.0.0.1:7000 doesn't have the object
127.0.0.1:7001 has the object (should be 3 copies)
127.0.0.1:7004 doesn't have the object
127.0.0.1:7005 doesn't have the object
127.0.0.1:7006 doesn't have the object
127.0.0.1:7007 has the object (should be 3 copies)
127.0.0.1:7008 doesn't have the object
127.0.0.1:7009 doesn't have the object
127.0.0.1:7010 doesn't have the object
127.0.0.1:7011 doesn't have the object
127.0.0.1:7012 has the object (should be 3 copies)
127.0.0.1:7013 doesn't have the object
127.0.0.1:7014 doesn't have the object
127.0.0.1:7015 doesn't have the object
127.0.0.1:7016 doesn't have the object
127.0.0.1:7017 doesn't have the object
127.0.0.1:7018 doesn't have the object
127.0.0.1:7019 doesn't have the object
127.0.0.1:7020 doesn't have the object
127.0.0.1:7021 doesn't have the object
127.0.0.1:7022 doesn't have the object
127.0.0.1:7023 doesn't have the object
127.0.0.1:7024 doesn't have the object
Looking for the object 0xfd2de300000001 (the inode vid 0xfd2de3 idx 1) with 23 nodes

127.0.0.1:7000 doesn't have the object
127.0.0.1:7001 has the object (should be 4 copies)
127.0.0.1:7004 doesn't have the object
127.0.0.1:7005 has the object (should be 4 copies)
127.0.0.1:7006 doesn't have the object
127.0.0.1:7007 doesn't have the object
127.0.0.1:7008 doesn't have the object
127.0.0.1:7009 doesn't have the object
127.0.0.1:7010 doesn't have the object
127.0.0.1:7011 doesn't have the object
127.0.0.1:7012 doesn't have the object
127.0.0.1:7013 doesn't have the object
127.0.0.1:7014 has the object (should be 4 copies)
127.0.0.1:7015 doesn't have the object
127.0.0.1:7016 doesn't have the object
127.0.0.1:7017 doesn't have the object
127.0.0.1:7018 doesn't have the object
127.0.0.1:7019 has the object (should be 4 copies)
127.0.0.1:7020 doesn't have the object
127.0.0.1:7021 doesn't have the object
127.0.0.1:7022 doesn't have the object
127.0.0.1:7023 doesn't have the object
127.0.0.1:7024 doesn't have the object
Looking for the object 0xfd2c3000000001 (the inode vid 0xfd2c30 idx 1) with 23 nodes

127.0.0.1:7000 has the object (should be 6 copies)
127.0.0.1:7001 has the object (should be 6 copies)
127.0.0.1:7004 doesn't have the object
127.0.0.1:7005 doesn't have the object
127.0.0.1:7006 doesn't have the object
127.0.0.1:7007 doesn't have the object
127.0.0.1:7008 doesn't have the object
127.0.0.1:7009 has the object (should be 6 copies)
127.0.0.1:7010 doesn't have the object
127.0.0.1:7011 has the object (should be 6 copies)
127.0.0.1:7012 doesn't have the object
127.0.0.1:7013 doesn't have the object
127.0.0.1:7014 doesn't have the object
127.0.0.1:7015 doesn't have the object
127.0.0.1:7016 doesn't have the object
127.0.0.1:7017 doesn't have the object
127.0.0.1:7018 doesn't have the object
127.0.0.1:7019 doesn't have the object
127.0.0.1:7020 doesn't have the object
127.0.0.1:7021 has the object (should be 6 copies)
127.0.0.1:7022 doesn't have the object
127.0.0.1:7023 has the object (should be 6 copies)
127.0.0.1:7024 doesn't have the object
Looking for the object 0xfd314900000001 (the inode vid 0xfd3149 idx 1) with 23 nodes

127.0.0.1:7000 doesn't have the object
127.0.0.1:7001 has the object (should be 7 copies)
127.0.0.1:7004 doesn't have the object
127.0.0.1:7005 doesn't have the object
127.0.0.1:7006 doesn't have the object
127.0.0.1:7007 doesn't have the object
127.0.0.1:7008 has the object (should be 7 copies)
127.0.0.1:7009 doesn't have the object
127.0.0.1:7010 doesn't have the object
127.0.0.1:7011 has the object (should be 7 copies)
127.0.0.1:7012 has the object (should be 7 copies)
127.0.0.1:7013 doesn't have the object
127.0.0.1:7014 doesn't have the object
127.0.0.1:7015 doesn't have the object
127.0.0.1:7016 doesn't have the object
127.0.0.1:7017 doesn't have the object
127.0.0.1:7018 doesn't have the object
127.0.0.1:7019 has the object (should be 7 copies)
127.0.0.1:7020 doesn't have the object
127.0.0.1:7021 doesn't have the object
127.0.0.1:7022 has the object (should be 7 copies)
127.0.0.1:7023 doesn't have the object
127.0.0.1:7024 has the object (should be 7 copies)
Looking for the object 0xfd2f9600000001 (the inode vid 0xfd2f96 idx 1) with 23 nodes

127.0.0.1:7000 has the object (should be 23 copies)
127.0.0.1:7001 has the object (should be 23 copies)
127.0.0.1:7004 has the object (should be 23 copies)
127.0.0.1:7005 has the object (should be 23 copies)
127.0.0.1:7006 has the object (should be 23 copies)
127.0.0.1:7007 has the object (should be 23 copies)
127.0.0.1:7008 has the object (should be 23 copies)
127.0.0.1:7009 has the object (should be 23 copies)
127.0.0.1:7010 has the object (should be 23 copies)
127.0.0.1:7011 has the object (should be 23 copies)
127.0.0.1:7012 has the object (should be 23 copies)
127.0.0.1:7013 has the object (should be 23 copies)
127.0.0.1:7014 has the object (should be 23 copies)
127.0.0.1:7015 has the object (should be 23 copies)
127.0.0.1:7016 has the object (should be 23 copies)
127.0.0.1:7017 has the object (should be 23 copies)
127.0.0.1:7018 has the object (should be 23 copies)
127.0.0.1:7019 has the object (should be 23 copies)
127.0.0.1:7020 has the object (should be 23 copies)
127.0.0.1:7021 has the object (should be 23 copies)
127.0.0.1:7022 has the object (should be 23 copies)
127.0.0.1:7023 has the object (should be 23 copies)
127.0.0.1:7024 has the object (should be 23 copies)
